src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURES = 5;
+    private static final int DEFAULT_NUMBER = 10;
+    private static final int MAX_NUMBER = 10;
+    private static final String DEFAULT_COLOR = "White";
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(MAX_FIGURES);
+        String color = colorSupplier.getRandomColor();
+
+        return switch (index) {
+            case 0 -> new Circle(color, random.nextInt(MAX_NUMBER));

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example, `int radius = random.nextInt(MAX_NUMBER);` and then pass `radius` to the constructor.